hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -368,21 +381,13 @@ private static void sleepForTime(int sleepTimeMs) {
   public JavaRDD<WriteStatus> updateLocation(JavaRDD<WriteStatus> writeStatusRDD, JavaSparkContext jsc,
       HoodieTable<T> hoodieTable) {
     final HBaseIndexQPSResourceAllocator hBaseIndexQPSResourceAllocator = createQPSResourceAllocator(this.config);
-    JavaRDD<WriteStatus> writeStatusResultRDD;
     setPutBatchSize(writeStatusRDD, hBaseIndexQPSResourceAllocator, jsc);
-    logger.info("multiPutBatchSize: before puts" + multiPutBatchSize);
+    logger.info("multiPutBatchSize: before hbase puts" + multiPutBatchSize);
     JavaRDD<WriteStatus> writeStatusJavaRDD = writeStatusRDD.mapPartitionsWithIndex(
         updateLocationFunction(), true);
-    // Forcing a spark action so HBase puts are triggered before releasing resources
-    if (this.config.getHBaseIndexShouldComputeQPSDynamically()) {
-      logger.info("writestatus count: " + writeStatusJavaRDD.count());
-      writeStatusResultRDD = writeStatusRDD;
-    } else {
-      writeStatusResultRDD = writeStatusJavaRDD;
-    }
-    // Release QPS resources as HBAse puts are done at this point
-    hBaseIndexQPSResourceAllocator.releaseQPSResources();
-    return writeStatusResultRDD;
+    // caching the index updated status RDD
+    writeStatusJavaRDD = writeStatusJavaRDD.persist(config.getWriteStatusStorageLevel());
```

Comment:
So this cache is for the read side and the other one after the updateIndex has happened ?

Reply:
Yes, this is to avoid(best effort) triggering index update for any chained RDD action after this.

Reply:
Can we call releaseResources here ? Ideal case is after persist, in most cases, the dag will not be retriggered, in the very small chance it is, the resources will be acquired again. After we do this, remove the close() method.

Reply:
The problem is that, resources are acquired at spark driver level and so re-triggering might not acquire resources. I agree that after persisting, it might only be a small chance but to make best effort, releasing resources from close() at the end is safe, wdyt ?